js-core/homeworks/homework-17/index.js

```diff
@@ -0,0 +1,117 @@
+// TASK 0
+const solution = arr => {
+	let newArr = [];
+	let lastElem = arr[arr.length - 1];
+	arr.pop();
+	let tmp = function(arr) {
+		let index;
+		let leader = arr[0];
+		for(let i = 0; i < arr.length; i++) {
+			if(arr[i] >= leader) {
+				leader = arr[i];
+				index = i;
+			};
+		};
+		newArr.push(leader);
+		arr.splice(0, (index + 1));
+		if(arr.length > 1) {
+			tmp(arr);
+		};
+	};
+	tmp(arr);
+	newArr.push(lastElem);
+	return newArr;
+};
+
+console.log(solution([16, 17, 4, 3, 5, 2])); // === [17, 5, 2]
+console.log(solution([4, 3, 7, 12, 6, 67, 5, 45, 34, 35, 2, 8])); // [67, 45, 35, 8]
+console.log(solution([12, 10, 12, 8, 7, 6])); // [12, 8, 7, 6]
+console.log(solution([1, 2, 3, 4, 5, 4])); // [5, 4]
+console.log(solution([12, 12, 12])); // [12, 12]
+
+
+//TASK1
+class Carousel {
+	constructor() {
+		this.arrImg = ["bestsportcar170.jpg", "Acura-NSX-2016-2017-02.jpg", "ferrari_667_171011.jpg"];
```

Comment:
Probably Carousel should not be worrying about picture names.

Our Carousel should only be responsible to know about
 - active element
 - changing elements left/right

That's all. And we have to apply it in several places in every app :) without any knowledge of pictures